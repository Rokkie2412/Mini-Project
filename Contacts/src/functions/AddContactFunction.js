import { launchCamera,launchImageLibrary } from "react-native-image-picker"

export const allowAddTrueFalse = (firstname,lastname,age,disable,setDisable) =>{
    if((firstname === '' || lastname === '') || age === ''){
        setDisable(true)
        return(disable)
    }else{
        setDisable(false)
        return(disable)
    }
}

export const addContactToAPI = (first,last,age,photo) =>{
    const postContact = fetch('https://simple-contact-crud.herokuapp.com/contact',{
        method:'POST',
        body:JSON.stringify({
            firstName : first,
            lastName : last,
            age : age,
            photo : photo
        }),
        headers:{
            'content-type' : 'application/json',
        }
    }).then((Response)=>Response.json())
    .then(json => console.log(json))

    return postContact
}

export const openCamera = (imageFromCamera,setImageFromCamera) => {
        
        const option={
            mediaType : 'photo',
            quality : 0.6,
        }
        launchCamera(option,(res)=>{
            if(res.didCancel){
                console.log('User Cancel Image Picker');
            }else if(res.errorCode){
                console.log(res.errorMessage);
            }else{
                const data = res.assets[0]
                setImageFromCamera(data.uri)
                console.log(data.uri)
            }
        })
        return imageFromCamera
    }

export const openGallery = (imageFromGallery,setImageFromGallery) =>{
    const option = {
        mediaType : 'photo',
        quality : 0.6
    }
    launchImageLibrary(option,(res)=>{
        if(res.didCancel){
            console.log('User Cancel Image Picker')
        }else if(res.errorCode){
            console.log(res.errorCode)
        }else{
            const data= res.assets[0]
            setImageFromGallery(data.uri)
            console.log(data.uri)
        }
    })
    return imageFromGallery
}