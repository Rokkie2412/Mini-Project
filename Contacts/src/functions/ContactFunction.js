export const GettingData = (Contact,setContact,setTempContact) =>{
    fetch('https://simple-contact-crud.herokuapp.com/contact')
    .then((response)=>response.json())
    .then((json)=>{
        setContact(json.data)
        setTempContact(json.data)
    })
    .catch((error)=>console.log(error))
    return Contact
}

export const FilterSearch = (text,Contact,setTempContact,setText) => {
    if(text){
        const newData = Contact.filter((item)=>{
            const itemData = item.firstName ? item.firstName.toUpperCase() : "".toUpperCase()
            const textData = text.toUpperCase()
            return itemData.indexOf(textData) > -1
        })
        setTempContact(newData)
        setText(text)
    }else{
        setTempContact(Contact)
        setText(text)
    }
}