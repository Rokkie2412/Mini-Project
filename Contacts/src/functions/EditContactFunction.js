export const editContactToAPI = (first,last,age,photo,id) =>{
    const putContact = fetch(`https://simple-contact-crud.herokuapp.com/contact/${id}`,{
        method:'PUT',
        body:JSON.stringify({
            firstName : first,
            lastName : last,
            age : age,
            photo : photo
        }),
        headers:{
            'content-type' : 'application/json',
        }
    }).then((Response)=>Response.json())
    .then(json => console.log(json))

    return putContact
}