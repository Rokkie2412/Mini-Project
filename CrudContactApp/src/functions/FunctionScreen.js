import axios from 'axios'
import * as React from 'react'
import NetInfo from "@react-native-community/netinfo";
import RNRestart from 'react-native-restart';
import { launchCamera,launchImageLibrary } from "react-native-image-picker"
import styles from '../styles/ContactAppStyle'
import {View,Text,KeyboardAvoidingView,FlatList,Pressable,Image} from 'react-native'

export const getData = (setData:Function,Data:Object,FilteredData:Object,setFilteredData:Function):Function =>{
    axios.get('https://simple-contact-crud.herokuapp.com/contact')
    .then(res=>{
    setFilteredData(res.data.data)
    setData(res.data.data)
    })
    return Data
}

export const addData = (first:string,last:string,age:number,foto:string):void =>{
    axios.post('https://simple-contact-crud.herokuapp.com/contact',{
        'firstName' : first,
        'lastName' : last,
        'age' : age,
        'photo' : foto
    },
    {
        headers:{
            'content-type' : 'application/json',
        }
    })
}

export const putData = (first:string,last:string,age:number,foto:string,id:string):void =>{
    axios.put(`https://simple-contact-crud.herokuapp.com/contact/${id}`,{
        'firstName' : first,
        'lastName' : last,
        'age' : age,
        'photo' : foto
    },
    {
        headers:{
            'content-type' : 'application/json',
        }
    })
}

export const FilterSearch = (text:string,Contact:Object,setTempContact:Function,setText:Function):void => {
    if(text){
        const newData = Contact.filter((item)=>{
            const itemData = item.firstName ? item.firstName.toUpperCase() : "".toUpperCase()
            const textData = text.toUpperCase()
            return itemData.indexOf(textData) > -1
        })
        setTempContact(newData)
        setText(text)
    }else{
        setTempContact(Contact)
        setText(text)
    }
}

export const openCamera = (image:string,setImage:Function):Function => {
        
        const option={
            mediaType : 'photo',
            quality : 0.6,
        }
        launchCamera(option,(res)=>{
            if(res.didCancel){
                console.log('User Cancel Image Picker');
            }else if(res.errorCode){
                console.log(res.errorMessage);
            }else{
                const data = res.assets[0]
                setImage(data.uri)
                console.log(data.uri)
            }
        })
        return image
    }

export const openGallery = (image:string,setImage:Function):Function =>{
    const option = {
        mediaType : 'photo',
        quality : 0.6
    }
    launchImageLibrary(option,(res)=>{
        if(res.didCancel){
            console.log('User Cancel Image Picker')
        }else if(res.errorCode){
            console.log(res.errorCode)
        }else{
            const data= res.assets[0]
            setImage(data.uri)
            console.log(data.uri)
        }
    })
    return image
}

export const AddorEditContactLock = (first:string,last:string,age:string):Function =>{
    let correctfirstname:boolean = false
    let correctlastname:boolean = false
    let correntage:boolean = false
    let errorText:string = ''
    let allowAdd:boolean = false

    if(first.length >= 3){
        correctfirstname = true
    }else if(first.length <3){
        correctfirstname = false
    }

    if(last.length >= 3){
        correctlastname = true
    }else if(last.length <3){
        correctlastname = false
    }

    if(last.length >= 3){
        correctlastname = true
    }else if(last.length <3){
        correctlastname = false
    }

    if(age.length !== 0 ){
        correntage = true
    }else if(last.length === 0 ){
        correntage = false
    }

    if(correctfirstname === true && correctlastname === true && correntage === true){
        allowAdd = true
    }else{
        allowAdd = false
    }

   return allowAdd
}

export const emptyAll = (setImage:Function,setLast:Function,setFirst:Function,setAge:Function):void =>{
    setImage('N/A')
    setLast('')
    setFirst('')
    setAge('')
}

export const FuncRepeat = (text:string,setRepeat:Function):void =>{
    if(text.length === 0){
        setRepeat(1000)
    }else{
        setRepeat(100000)
    }
}

export const CheckInternet = (set:Function):void =>{
    NetInfo.fetch('https://simple-contact-crud.herokuapp.com/contact')
    .then(networkState=>{
        if(networkState.isConnected !== true){
            set(true)
        }
    })
}
export const AddPack = (setmodal:Function,age:string,first,last,image,setImage,setLast,setFirst,setAge,setError):void =>{
        setmodal(false)
        const tempPasre = parseInt(age)
        addData(first,last,tempPasre,image)
        setError('')
        emptyAll(setImage,setLast,setFirst,setAge)
    }

export const Restart = ():void =>{
    RNRestart.Restart()
}
       