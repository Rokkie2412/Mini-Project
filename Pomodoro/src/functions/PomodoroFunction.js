import reactNativeBackgroundTimer from 'react-native-background-timer';

export const displayTimer = (secondLeft) =>{
    let hours = Math.floor(secondLeft/60/60)
    let minutes = Math.floor(secondLeft/60%60)
    let seconds = Math.floor(secondLeft%60)

    let displayHours = hours < 10 ? `0${hours}` : hours
    let displayMins = minutes < 10 ? `0${minutes}` : minutes
    let displaySecs = seconds < 10 ? `0${seconds}` : seconds

    return{
      displayHours,
      displayMins,
      displaySecs
    }
}

export const startTimer = (setSecondLeft) =>{
  reactNativeBackgroundTimer.runBackgroundTimer(()=>{
    setSecondLeft(secs =>{
      if(secs > 0){
        return secs-1
      }else if (secs === 0 ) {
        reactNativeBackgroundTimer.stopBackgroundTimer()
      }
    })
  },1000)
}

export const onchecklimit = (value,setValue) =>{
    const parsedqty = Number.parseInt(value)
    if(Number.isNaN(parsedqty))
      setValue(0)
    else if( parsedqty >59)
      setValue(59)
    else
      setValue(parsedqty)
  }