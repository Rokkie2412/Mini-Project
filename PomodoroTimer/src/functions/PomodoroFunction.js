import reactNativeBackgroundTimer from 'react-native-background-timer';
import PushNotification from "react-native-push-notification";
import { Vibration } from 'react-native';
export const displayTimer = (secondLeft) =>{
    let hours = Math.floor(secondLeft/60/60)
    let minutes = Math.floor(secondLeft/60%60)
    let seconds = Math.floor(secondLeft%60)

    let displayHours = hours < 10 ? `0${hours}` : hours
    let displayMins = minutes < 10 ? `0${minutes}` : minutes
    let displaySecs = seconds < 10 ? `0${seconds}` : seconds

    return{
      displayHours,
      displayMins,
      displaySecs
    }
}

export const startTimer = (setSecondLeft) =>{
  reactNativeBackgroundTimer.runBackgroundTimer(()=>{
    setSecondLeft(secs =>{
      if(secs > 0){
        return secs-1
      }else if (secs === 0 ) {
        reactNativeBackgroundTimer.stopBackgroundTimer()
      }
    })
  },1000)
}

export const onchecklimit = (value,setValue) =>{
    const parsedqty = Number.parseInt(value)
    if(Number.isNaN(parsedqty))
      setValue(0)
    else if( parsedqty >59)
      setValue(59)
    else
      setValue(parsedqty)
  }

export const createChannel = () =>{
    PushNotification.createChannel({
      channelId:'test-channel',
      channelName:'Test Channel',
      vibrate:true
    })
  }

export const handleNotification = (judul,pesan) =>{
    PushNotification.localNotification({
      channelId:"test-channel",
      title:judul,
      message:pesan,
      playSound:true,
      id:1,
      vibrate:true,
      vibration:1000
    })
  }
  
export const handleNotificationLongBreak = () =>{
    PushNotification.localNotification({
      channelId:"test-channel",
      title:'Time to long break!',
      message:"it's time to long break! Use it wisely",
      playSound:true,
      id:1,
      vibrate:true,
      vibration:1000,
    })
  }

export const TimerinNotification = (judul,pesan) =>{
    PushNotification.localNotification({
      channelId:"test-channel",
      title:judul,
      message:pesan,
      playSound:false,
      id:2,
      vibrate:false,
    })
  }