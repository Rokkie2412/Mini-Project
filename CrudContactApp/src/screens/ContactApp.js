/**
 * @flow
 */

import * as React from 'react'
import {View,Text,KeyboardAvoidingView,FlatList,Pressable,Image} from 'react-native'
import styles from '../styles/ContactAppStyle'
import {useEffect,useState} from 'react'
import {getData,FilterSearch,FuncRepeat,CheckInternet,Restart} from '../functions/FunctionScreen'
import Icon from 'react-native-vector-icons/Ionicons'
import AddContactModal from '../components/AddContactModal'
import Searchbar from '../components/Searchbar'
import ContactCard from '../components/ContactCard'
import EditContact from '../components/EditContactModal'
import RNRestart from 'react-native-restart';
import AwesomeAlert from 'react-native-awesome-alerts'

const ContactApp = ({navigation}:Object):React.Node =>{
    const [addContact,setAddContact] = useState<boolean>(false)
    const [arrayList,setArrayList] = useState<Object>([])
    const [showCard,setShowCard] = useState<boolean>(false)
    const [showEdit,setShowEdit] = useState<boolean>(false)
    const [id,setId] = useState<string>('')
    const [first,setFirst] = useState<string>('')
    const [last,setLast] = useState<string>('')
    const [age,setAge] = useState<string>('')
    const [image,setImage] = useState<string>('')
    const [filteredData,setFilterData] = useState<Object>([])
    const [search,setSearch] = useState<string>('')
    const [alert,setAlert] = useState<boolean>(false)

    useEffect(()=>{
        getData(setArrayList,arrayList,filteredData,setFilterData)
    },[addContact,showEdit])
    
    useEffect(() => {
    const interval = setInterval(() => {
        CheckInternet(setAlert)
    }, 2000);
    return () => clearInterval(interval);
    },[]);

    const _render = ({item}) =>{
        return(
            <View style={styles.flatcontainer}>   
                <Pressable
                testID='FlatlistButton'
                onPress={()=>{
                setId(item.id)
                setFirst(item.firstName)
                setLast(item.lastName)
                setAge(item.age)
                setImage(item.photo)
                setShowCard(true)
                }}
                onLongPress={()=>{
                setId(item.id)
                setFirst(item.firstName)
                setLast(item.lastName)
                setAge(item.age)
                setImage(item.photo)
                setShowEdit(true)
                }}
                style={styles.contactList}>
                    {item.photo === "N/A" ? <Icon style={styles.personIcon} name="person"/> :
                    <Image style={styles.imagesPerson} source={{uri:`${item.photo}`}}/>
                    }
                    <Text style={styles.nameList}>{item.firstName} {item.lastName}</Text>
                    </Pressable>
            </View>
        )
    }

    return(
        <KeyboardAvoidingView 
        style={styles.mainContainer}>
            <View style={styles.HeaderView}>
                <Text style={styles.Header}>Contact</Text>
            </View>
            <Searchbar search={search} setSearch={(text)=>FilterSearch(text,arrayList,setFilterData,setSearch)}/>
            <View style={styles.body}>
                <FlatList
                    testID='Flastlist'
                    showsVerticalScrollIndicator={false}
                    data={filteredData.sort((a,b)=>a.firstName.localeCompare(b.firstName))}
                    keyExtractor={({id},index)=>id}
                    renderItem={_render}
                />
            </View>
            <Pressable 
            testID='addContactButton'
            onPress={()=>{
                setAddContact(true)
            }}
            style={styles.addContainer}>
                <Icon name="add" style={styles.addIcon}/>
            </Pressable>
            <AddContactModal showmodal={addContact} setmodal={setAddContact}/>
            <ContactCard show={showCard} setshow={setShowCard} image={image} first={first} last={last} Age={age}/>
            <EditContact 
            show={showEdit} setshow={setShowEdit} 
            id={id} first={first} last={last} 
            age={age} photo={image}
            setfirst={setFirst}
            setlast={setLast}
            setage={setAge}
            setphoto={setImage}
            />
            <AwesomeAlert
                show={alert}
                title="Cannot connected to internet"
                message='Please check internet connection and restart application'
                closeOnTouchOutside={false}
                closeOnHardwareBackPress={false}
                showConfirmButton={true}
                confirmText="Restart application"
                confirmButtonColor="#00ADB5"
                onConfirmPressed={Restart}
            />
        </KeyboardAvoidingView>
    )
}

export default ContactApp

