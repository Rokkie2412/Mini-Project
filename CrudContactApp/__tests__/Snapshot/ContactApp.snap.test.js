import {create,act,renderer} from 'react-test-renderer';
import React from 'react';
import ContactApp,{_render} from "../../src/screens/ContactApp";
import { FlatList } from 'react-native';
import {CheckInternet} from '../../src/functions/FunctionScreen'
import {
  render,
  waitForElement,
  fireEvent,
} from 'react-native-testing-library';

//Mocking Session
jest.useFakeTimers();
jest.mock('../../src/functions/FunctionScreen',()=>({
    FilterSearch : jest.fn(),
    getData:jest.fn()
}))
jest.mock('../../src/components/Searchbar',()=> 'Searchbar')
jest.mock('../../src/components/AddContactModal',()=> 'AddContactModal')
jest.mock('../../src/components/ContactCard',()=>'ContactCard')
jest.mock('../../src/components/EditContactModal',()=>'EditContact')
jest.mock('react-native-vector-icons/Ionicons', () => 'Icon');
jest.mock('react-native-restart',()=>{
    return{
        Restart:jest.fn()
    }
})
jest.mock('react-native-awesome-alerts',()=>'AwesomeAlert')
jest.mock('@react-native-community/netinfo',()=>'Netinfo')
jest.mock('react-native-image-picker',()=>{
    return{
        launchCamera:jest.fn(),
        launchImageLibrary:jest.fn()
    }
})



//Testing Session
const tree = create(<ContactApp/>)

describe('Testing for ContactApp Screen',()=>{
    describe('Snapshot test for ContactApp',()=>{
        it('Should render corretly',()=>{
            const tree = create(<ContactApp/>)
            expect(tree.toJSON()).toMatchSnapshot()
        })
    })
    describe('testing button for contactApp',()=>{
        it('Should get pressed for add button',()=>{
            const tree = create(<ContactApp/>)
            const button = tree.root.findByProps({testID:'addContactButton'}).props
            act(()=>button.onPress())
        })
    })
    
    describe('Flastlist',()=>{
        it('Should render corretly',()=>{
            const _render = jest.fn()
            expect(renderer.create(
                <FlatList
                    data={['foo']}
                    keyExtractor={({id},index)=>id}
                    renderItem={_render}
                />
            )).toMatchSnapshot()
        })
    })

    
})
