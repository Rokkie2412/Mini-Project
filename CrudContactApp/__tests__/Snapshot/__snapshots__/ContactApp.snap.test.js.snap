// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing for ContactApp Screen Snapshot test for ContactApp Should render corretly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "backgroundColor": "#222831",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#EEEEEE",
          "fontSize": 32,
          "fontWeight": "bold",
          "margin": "4%",
        }
      }
    >
      Contact
    </Text>
  </View>
  <Searchbar
    search=""
    setSearch={[Function]}
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      data={Array []}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      testID="Flastlist"
      viewabilityConfigCallbackPairs={Array []}
    >
      <View />
    </RCTScrollView>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={Object {}}
    testID="addContactButton"
  >
    <Icon
      name="add"
      style={
        Object {
          "alignSelf": "flex-end",
          "backgroundColor": "#44865a",
          "borderRadius": 50,
          "bottom": 0,
          "color": "#EEEEEE",
          "fontSize": 40,
          "margin": 10,
          "padding": 15,
          "position": "absolute",
        }
      }
    />
  </View>
  <AddContactModal
    setmodal={[Function]}
    showmodal={false}
  />
  <ContactCard
    Age=""
    first=""
    image=""
    last=""
    setshow={[Function]}
    show={false}
  />
  <EditContact
    age=""
    first=""
    id=""
    last=""
    photo=""
    setage={[Function]}
    setfirst={[Function]}
    setlast={[Function]}
    setphoto={[Function]}
    setshow={[Function]}
    show={false}
  />
  <AwesomeAlert
    closeOnHardwareBackPress={false}
    closeOnTouchOutside={false}
    confirmButtonColor="#00ADB5"
    confirmText="Restart application"
    message="Please check internet connection and restart application"
    show={false}
    showConfirmButton={true}
    title="Cannot connected to internet"
  />
</View>
`;
